src/main/java/org/lsc/utils/ScriptingEvaluator.java

```diff
@@ -90,7 +91,12 @@ else if ("graal.js".equals(name)) {
 			defaultImplementation = graaljsevaluator;
 		}
 		else {
-			defaultImplementation = instancesTypeCache.get("js");
+			for ( String name: new String[] {"js","rjs"} ) {
+				defaultImplementation = instancesTypeCache.get(name);
+				if ( defaultImplementation != null ) {
+					break;
+				}
+			}
```

Comment:
If understand well, this change means if you don't find any "js" implementation, you defaults on "rjs" which is in fact Rhino.
It's hard to understand that when reading your code at first sight. Wouldn't it be possible to make it more obvious?

Reply:
yes you understand well, trying to get the better default implementation.

it might be possible to make it more obvious, but making things obvious is not as obvious as it sound ;-)
Does just a comment would help ?
Or {"js","rjs"} set in a variable named 'preferredDefaultImplementations' ?

Reply:
I think I would just be more explicit, even if it means repeating myself:

```java
defaultImplementation = instancesTypeCache.get("js");
if (defaultImplementation == null) {
  defaultImplementation = instancesTypeCache.get("rjs");
}
```
with the optional solution it's even easier to write it:
```java
			defaultImplementation = Optional.ofNullable(instancesTypeCache.get("js"))
					.or(() -> Optional.ofNullable(instancesTypeCache.get("rjs")));
```
It's very easy and clear (to me at least ^^) how to add new cases if needed later (I guess it's the purpose of your array).